dappnode/DNP_DAPPMANAGER
# Improve + eth2MigrationRequirements`

## Context

The upcoming eth2 migration for gnosis chain must ensure all the possible edge cases

## Approach

Implement restrictions and tests for the following cases:
 - Ensure not installing Prysm legacy
 - Ensure not installing web3signer if Prysm legacy is installed
 - Ensure not installing any other client if Prysm legacy is installed

## Test instructions (having dappmanager with this PR code)

Prysm gnosis legacy version  <= 0.1.8 

- Ensure not installing Prysm legacy:  make sure it is not allowed to install Prysm gnosis legacy
- Ensure not installing web3signer if Prysm legacy is installed:  install Prysm gnosis legacy before this dappmanager (otherwise it should not be allowed). Then try to install web3signer-gnosis
- Ensure not installing any other client if Prysm legacy is installed:  install Prysm gnosis legacy before this dappmanager (otherwise it should not be allowed). Then try to install any other client-gnosis.
- Ensure it is allowed to install any other package!!
